client/components/swipeable/index.js

```diff
@@ -0,0 +1,212 @@
+
+	const pagesRef = useRef();
+	const resizeObserverRef = useRef();
+	const numPages = Children.count( children );
+
+	const getWidth = () => {
+		return resizeObserverRef.current?.getBoundingClientRect().width;
+	};
+
+	const getPagesWidth = () => {
+		return getWidth() * numPages;
+	};
+
+	const getOffset = ( index ) => {
+		return -( getWidth() * index );
+	};
+
+	function useUpdateLayout( enabled, currentPageIndex, updateLayout ) {
+		// save callback to a ref so that it doesn't need to be a dependency of other hooks
+		const savedUpdateLayout = useRef();
+		useEffect( () => {
+			savedUpdateLayout.current = updateLayout;
+		}, [ updateLayout ] );
+
+		// fire when the `currentPageIndex` parameter changes
+		useEffect( () => {
+			if ( ! enabled ) {
+				return;
+			}
+
+			savedUpdateLayout.current();
+		}, [ enabled, currentPageIndex ] );
+
+		// fire when the window resizes
+		useEffect( () => {
+			if ( ! enabled ) {
+				return;
+			}
+
+			const onResize = () => savedUpdateLayout.current( false );
+			window.addEventListener( 'resize', onResize );
+			window.addEventListener( 'orientationchange', onResize );
+			return () => {
+				window.removeEventListener( 'resize', onResize );
+				window.removeEventListener( 'orientationchange', onResize );
+			};
+		}, [ enabled ] );
+	}
+
+	const updateEnabled = hasDynamicHeight && numPages > 1;
+
+	useUpdateLayout( updateEnabled, currentPage, ( hasTransitionDuration = true ) => {
+		const offset = getOffset( currentPage );
+		const transform = {
+			transform: `translate3d(${ offset }px, 0px, 0px)`,
+			transitionDuration: hasTransitionDuration ? `300ms` : `0ms`,
+		};
+
+		let height = { height: null };
+		const targetHeight = pagesRef.current?.querySelector( '.is-current' )?.offsetHeight;
+
+		if ( targetHeight && pagesStyle?.height !== targetHeight ) {
+			height = { height: null };
+		}
+
+		setPagesStyle( { ...pagesStyle, ...height, ...transform } );
+	} );
+
+	useEffect( () => {
+		if ( currentPage >= numPages ) {
+			onPageSelect( numPages - 1 );
+		}
+	}, [ numPages, currentPage, onPageSelect ] );
+
+	const getDragPosition = ( event ) => {
+		if ( event.hasOwnProperty( 'clientX' ) ) {
+			return { x: event.clientX, y: event.clientY };
+		}
+
+		if ( event.targetTouches[ 0 ] ) {
+			return { x: event.targetTouches[ 0 ].clientX, y: event.targetTouches[ 0 ].clientY };
+		}
+
+		const touch = event.changedTouches[ 0 ];
+		return { x: touch.clientX, y: touch.clientY };
+	};
+
+	const handleDragStart = ( event ) => {
+		const position = getDragPosition( event );
+		setDragStartData( position );
+		setPagesStyle( { ...pagesStyle, transitionDuration: `0ms` } ); // Set transition Duration to 0 for smooth dragging.
+		setIsDragging( true );
+	};
+
+	const handleDragEnd = ( event ) => {
+		if ( ! isDragging ) {
+			return; // End early if we are not dragging any more.
+		}
+
+		const THRESHOLD_OFFSET = 100; // Number of pixels to travel before we trigger the slider to move to the desired slide.
```

Comment:
Should this be defined at the very top of the file (after the imports), as a constant?

Reply:
I like when constants are defined where they are used when they are only used once. 
But if you want to see this change I am open to it.  :)

Reply:
My main point about this is that this constant would be re-defined every time the `handleDragEnd` function runs.

Anyway it's definitely not a big deal, feel free to keep it as you prefer.